app/models/task.py

```diff
@@ -1,5 +1,40 @@
 from app import db
-
+from flask import abort, make_response
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(32), nullable=False)
+    description = db.Column(db.String(120), nullable=False)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.id'), nullable=True)
+
+    def to_json(self):
+        task = {
+            "id": self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": True
+        }
+        
+        if not self.goal_id and not self.completed_at:
+            task["is_complete"] = False
+        
+        elif self.goal_id:
+            task["goal_id"] = self.goal_id
+            if self.completed_at == None:
+                task["is_complete"] = False
+
+        return task
+
+    @classmethod
+    def from_json(cls, response_data):
+        try:
+            task = cls(
+                title = response_data["title"],
+                description = response_data["description"]
+                )
+        except KeyError:
```

Comment:
Like we validated models, could we also make a general function that validates all request bodies? Here's an example:
```python
def validate_request_body(request_body, keys):
    for key in keys:
        if not request_body.get(key): 
            abort(make_response({
                'Invalid Data': f'missing key: {key}'
            }, 400))

    return True
```
We can pass in the `request_body` and a list of strings that are `keys` and then check to see if those keys are present.